### PR TITLE
[ Term Entry ] PHP String Functions Str_ends_with() adresses #2433

### PR DESCRIPTION
**Pull request by MMMMMMMZ**
### Description

added term str_end_with addressing issue #2433 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by MMMMMMMZ** [comment]
ok I fixed some problems with the checks and removed some weird spacing

File: content/php/concepts/string-functions/terms/str-ends-with/str-ends-with.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+This example is runnable and uses the **`str_ends_with()`** function:
+
+> **Note 1:** The second echo statement will be true as all strings end with empty strings.
+> **Note 2:** The third echo statement will be false because the case does not match.
+
+```codebyte/php
+
+<?php
+if (str_ends_with('NiceJob', 'Bob')) {
+    echo "NiceJob Ends with Bob\n"; // doesn't print
```
Comment (dakshdeepHERE) [inline content/php/concepts/string-functions/terms/str-ends-with/str-ends-with.md:64]:
```suggestion
    echo "NiceJob Ends with Bob\n"; 
```

File: content/php/concepts/string-functions/terms/str-ends-with/str-ends-with.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+}
+if (str_ends_with('NiceJob', '')) {
+    echo "All Strings end with empty String\n"; // does print due to the reason in note 1
```
Comment (dakshdeepHERE) [inline content/php/concepts/string-functions/terms/str-ends-with/str-ends-with.md:67]:
```suggestion
    echo "All Strings end with empty String\n"; 
```

File: content/php/concepts/string-functions/terms/str-ends-with/str-ends-with.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+}
+if (str_ends_with('NiceJob', 'job')) {
+    echo "str_ends_with() doesn't care about Case\n"; // doesn't print due to the reason in note 2
```
Comment (dakshdeepHERE) [inline content/php/concepts/string-functions/terms/str-ends-with/str-ends-with.md:70]:
```suggestion
    echo "str_ends_with() doesn't care about Case\n"; 
```

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
Few changes and also don't comments in codebyte

File: content/php/concepts/string-functions/terms/str-ends-with/str-ends-with.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
+---
+Title: 'str_ends_with()'
+Description: 'Checks if a string ends with a given substring.'
+Subjects:
+  - 'Computer Science'
+  - 'Web Design'
+  - 'Web Development'
```
Comment (caupolicandiaz) [inline content/php/concepts/string-functions/terms/str-ends-with/str-ends-with.md:7]:
```suggestion
```

File: content/php/concepts/string-functions/terms/str-ends-with/str-ends-with.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+Tags:
+  - 'Strings'
+  - 'Functions'
+CatalogContent:
+  - 'learn-php'
+  - 'paths/computer-science'
+---
+
+The **`str_ends_with()`** function Checks if a **`string`** ends with a given **`substring`**
```
Comment (caupolicandiaz) [inline content/php/concepts/string-functions/terms/str-ends-with/str-ends-with.md:16]:
```suggestion
The **`str_ends_with()`** function checks if a string ends with a given substring.
```

File: content/php/concepts/string-functions/terms/str-ends-with/str-ends-with.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+## Syntax
+
+```pseudo
+str_ends_with( $string , $substring )
+```
+
+The **`str_ends_with()`** function has two required parameters:
```
Comment (caupolicandiaz) [inline content/php/concepts/string-functions/terms/str-ends-with/str-ends-with.md:24]:
```suggestion
The `str_ends_with()` function has two required parameters:
```

Comment (caupolicandiaz) [reply content/php/concepts/string-functions/terms/str-ends-with/str-ends-with.md:24]: boldface is only used for the first instance of the function/concept/etc

File: content/php/concepts/string-functions/terms/str-ends-with/str-ends-with.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+- **`$string`**: Specifies the **`string`** to check if it ends with the **`$substring`**.
```
Comment (caupolicandiaz) [inline content/php/concepts/string-functions/terms/str-ends-with/str-ends-with.md:26]: remove the boldface formatting from the other terms in this entry

File: content/php/concepts/string-functions/terms/str-ends-with/str-ends-with.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+- **`$substring`**: Specifies the **`string`** to check for in the **`$string`** .
+
+The **`str_ends_with()`** function returns a **`bool`** which is **`true`** if **`$string`** ends with **`$substring`**, **`false`** otherwise.
+
+## Example
+
+The following example uses the **`str_ends_with()`** function to check if the `"fiddlesticks"` ends with `"sticks"` and then prints to the console `" fiddlesticks end with sticks "`:
```
Comment (caupolicandiaz) [inline content/php/concepts/string-functions/terms/str-ends-with/str-ends-with.md:33]:
```suggestion
The following example uses the **`str_ends_with()`** function to check if the `"fiddlesticks"` ends with `"sticks"` and then prints to the console:
```

File: content/php/concepts/string-functions/terms/str-ends-with/str-ends-with.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+
+```php
+
```
Comment (caupolicandiaz) [inline content/php/concepts/string-functions/terms/str-ends-with/str-ends-with.md:36]: remove empty lines in the code block

File: content/php/concepts/string-functions/terms/str-ends-with/str-ends-with.md [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+if (str_ends_with('fiddlesticks', 'sticks')) {
+echo " fiddlesticks end with sticks ";
+}
+?>
+
+```
+
+The example will result in the following output:
+
+```shell
```
Comment (caupolicandiaz) [inline content/php/concepts/string-functions/terms/str-ends-with/str-ends-with.md:47]: same as above

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@MMMMMMMZ I've completed my initial review. Please update the entry, and tag me when you've finished making the changes. Thanks

**Comment by MMMMMMMZ** [comment]
@caupolicandiaz I finished the Requested Changes and Tagged you as you wished.

**Review by SSwiniarski — APPROVED** [review]
@MMMMMMMZ Looks good to merge.

**Comment by SSwiniarski** [comment]
@MMMMMMMZ Congrats on the new entry! Live link here: https://www.codecademy.com/resources/docs/php/string-functions/str-ends-with

image

**Pull request merged**